File: typescript/app/src/engine/move.ts
export interface MoveData {
  name: string
  type: string
  category: string
  power: number
  accuracy: number
  source: string
}

export class Move {
  name: string
  type: string
  category: string
  power: number
  acc: number
  effectivePower: number
  turns: number
  tags: string[]

  constructor(data: MoveData) {
    this.name = data.name
    this.type = data.type
    this.category = data.category
    this.power = data.power
    this.acc = data.accuracy
    this.effectivePower = data.power
    this.turns = 1
    this.tags = []
  }
}

File: move.py
class Move:
    def __init__(self, *, name, move_type, category, power, acc):
        self.name = name
        self.type = move_type
        self.category = category
        self.power = power
        self.acc = acc
        self.effective_power = float(power)
        self.turns = 1
        self.tags: list[str] = []

    def __repr__(self):
        return f"Move({self.name}, {self.type}, {self.category}, {self.power})"
